src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T):GroupsMap<T> {
+  const groupByObject:GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const myKey = String(item[key]);
+
+    groupByObject[myKey] = items.filter((person) => item[key] === person[key]);
```

Comment:
In this case, you will reassign your key a couple of times with the same values.
Let's check if that key exist and push values to the according key